### PR TITLE
To get the __len__, should pass self._current_dataset_name to _valida…

### PR DESCRIPTION
**Pull request by limberc**
Pass self._current_dataset_name to self._validate_dataset_name. Or it would get error when calling __len__.

### CODE REVIEW

**Comment by gpengzhi** [comment]
Thank you very much for contributing to `texar-pytorch`! Could you fix the CI error at first? It seems to be a pylint error.
'''
************* Module texar.torch.data.data.data_iterators
texar/torch/data/data/data_iterators.py:559:0: C0301: Line too long (91/80) (line-too-long))
'''

File: texar/torch/data/data/data_iterators.py [hunk 1]
```diff
@@ -95,6 +94,7 @@ def _move_fn(x):
     from texar.torch.data.data.data_iterators_utils import \
         TexarMultiProcessingDataLoaderIter as _MultiProcessingDataLoaderIter
 
+
```
Comment (huzecong) [inline texar/torch/data/data/data_iterators.py:97]: These whitespace changes are probably why the CI lint failed. Let's get rid of these and only have changes relevant to the issue in this PR.

**Review by huzecong — APPROVED** [review]
Thanks for the contribution! I think this is a good fix. Please fix the CI issues before merging!

Related but definitely out of the scope for this PR: down the road we should probably refactor the data interfaces a bit, from hindsight the current implementation is not very intuitive.

File: texar/torch/data/data/data_iterators.py (hunk 1 above)
Comment (limberc) [reply texar/torch/data/data/data_iterators.py:97]: Wow, thanks for the advice.

**Comment by limberc** [comment]
I have deleted the whitespace according to @huzecong

File: texar/torch/data/data/data_iterators.py [hunk 2]
```diff
@@ -549,7 +555,7 @@ def __iter__(self) -> Iterator[Batch]:
         return self.get_iterator()
 
     def __len__(self):
-        return len(self._datasets[self._validate_dataset_name(None)])
+        return len(self._datasets[self._validate_dataset_name(self._current_dataset_name)])
```
Comment (huzecong) [inline texar/torch/data/data/data_iterators.py:558]: Sorry but I don't think you've solved all of the lint errors. Can you only keep the change on this line?

Also, I think this line is too long. Our line limit is 80 characters. Could you wrap this?

**Review by huzecong — CHANGES_REQUESTED** [review]

**Comment by huzecong** [comment]
This was addressed in #351, so I'm closing this PR. Thanks anyway for your contribution!

**Pull request closed without merging**